src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1530,12 +1530,20 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	public undo(): void {
 		if (this.undoService.canUndo(this.notebookUri)) {
 			this.undoService.undo(this.notebookUri);
+			const azdata_notebook_guid: string = this.getMetaValue('azdata_notebook_guid');
+			this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.UndoCell)
+				.withAdditionalProperties({ azdata_notebook_guid: azdata_notebook_guid })
```

Comment:
Let's make this easy and add a helper function that takes a NotebookModel (`this` in this case) and adds in the appropriate Notebook-related telemetry (currently just the guid for now but more may be added later such as language).

(and update the existing events that do a similar thing to use this new function)